homeassistant/components/powerwall/sensor.py

```diff
@@ -32,17 +33,22 @@
 _METER_DIRECTION_EXPORT = "export"
 _METER_DIRECTION_IMPORT = "import"
 
+_ValueParamT = TypeVar("_ValueParamT")
+_ValueT = TypeVar("_ValueT", bound=float)
+
 
 @dataclass(frozen=True)
-class PowerwallRequiredKeysMixin:
+class PowerwallRequiredKeysMixin(Generic[_ValueParamT, _ValueT]):
     """Mixin for required keys."""
 
-    value_fn: Callable[[MeterResponse], float]
+    value_fn: Callable[[_ValueParamT], _ValueT]
```

Comment:
Do we have anything that doesn't use MetersResponse and float or is that coming in a future PR?

Reply:
Yep, right [here](https://github.com/home-assistant/core/pull/108339/files#diff-dcae9376379c1150d36869c466596805b70cd5f3faad8f1bedc17de5f43e824fR324).